src/Compilers/CSharp/Portable/Symbols/SymbolWithAnnotations.cs

```diff
@@ -752,11 +757,10 @@ public ReferenceTypeUnknownNullabilityWithoutCustomModifiers(TypeSymbol typeSymb
 
             public override TypeSymbol AsTypeSymbolOnly()
             {
-                Debug.Assert(this.IsNullable == false);
                 return _typeSymbol;
             }
 
-            public sealed override bool Is(TypeSymbol other) => false; // It has nullable annotation, unknown is an annotation.
+            public sealed override bool Is(TypeSymbol other) => _typeSymbol.Equals(other, TypeCompareKind.CompareNullableModifiersForReferenceTypes);
```

Comment:
> _typeSymbol.Equals(other, TypeCompareKind.CompareNullableModifiersForReferenceTypes); [](start = 63, length = 86)

As we discussed offline this means we need to adjust implementation of this method in the class that represents not-nullable reference types.

Reply:
I tried that change, along with using `IsNullable=null` for `TypeParameters` as `TypeArguments`, and it will be a significant change with an unknown amount of work to get all tests passing. I'd like to postpone that to a separate PR.

---
In reply to: [171438350](https://github.com/dotnet/roslyn/pull/25087#discussion_r171438350) [](ancestors = 171438350)